### PR TITLE
Stop command

### PR DESCRIPTION
**Pull request by Saud-97**
This will work for all cases including when 247 is enabled compared to the previous stop command PR.

### CODE REVIEW

**Comment by XstreamSpeed** [comment]
We already have a disconnect command which practically does the same thing, but this time leaves the voice channel. I see this as kind of an extra feature but kind of not, it has its pros and cons.

**Comment by Saud-97** [comment]
> We already have a disconnect command which practically does the same thing, but this time leaves the voice channel. I see this as kind of an extra feature but kind of not, it has its pros and cons.

how does **leave** work with 24/7 enabled? It doesn't make sense to even call it leave if it shouldn't leave.

**Comment by ItzCuteAkemi** [comment]
hmm, well this can be useful if you don't want the bot connect and disconnect from voice channel.
@SudhanPlayz thoughs?

**Comment by DevAmiyo** [comment]
wait.. doesn't the bot leave with /leave when 24/7 is enabled?

**Comment by BioCla** [comment]
> wait.. doesn't the bot leave with /leave when 24/7 is enabled?

Yes, that's the intended behavior

**Comment by ItzCuteAkemi** [comment]
however it's literally same commands so yea makes no sense

File: commands/slash/stop.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        embeds: [
+          new MessageEmbed()
+            .setColor("RED")
+            .setDescription("Nothing is playing right now."),
+        ],
+        ephemeral: true,
+      });
+    }
+
+    player.stop();
```
Comment (BioCla) [inline commands/slash/stop.js:34]: This should be switched with line 35

**Comment by Saud-97** [comment]
What I'm saying is that there is no command to clear and stop the bot while using 24/7. You must use skip and clear command to do the same as this stop command.

**Comment by DevAmiyo** [comment]
> What I'm saying is that there is no command to clear and stop the bot while using 24/7. You must use skip and clear command to do the same as this stop command.

Ohhhh, that's what you wanted. I see. Maybe this functionality can be added via buttons in the currently playing thing?
That way a duplicate command isn't created and the feature also stays. @BioCla thoughts?

**Comment by BioCla** [comment]
> > What I'm saying is that there is no command to clear and stop the bot while using 24/7. You must use skip and clear command to do the same as this stop command.
> 
> Ohhhh, that's what you wanted. I see. Maybe this functionality can be added via buttons in the currently playing thing? That way a duplicate command isn't created and the feature also stays. @BioCla thoughts?

We actually discussed this in the Discord server earlier today, came up with:
```js
if (player.twentyFourSeven) {
  player.queue.clear();
  player.stop();
} else {
  player.destroy();
}
```

**Comment by DevAmiyo** [comment]
> We actually discussed this in the Discord server earlier today, came up with:
> 
> ```js
> if (player.twentyFourSeven) {
>   player.queue.clear();
>   player.stop();
> } else {
>   player.destroy();
> }
> ```

Yeah.. this works too. So, this stays in the `/stop` command ?

**Comment by BioCla** [comment]
> > We actually discussed this in the Discord server earlier today, came up with:
> > 
> > ```js
> > if (player.twentyFourSeven) {
> >   player.queue.clear();
> >   player.stop();
> > } else {
> >   player.destroy();
> > }
> > ```
> 
> Yeah.. this works too. So, this stays in the `/stop` command ?

It can go in the `/leave` command, and in each case maybe send a different embed. In the first case for example:
"24/7 active, queue has been cleared and player stopped"

**Comment by Saud-97** [comment]
> It can go in the `/leave` command, and in each case maybe send a different embed. In the first case for example: "24/7 active, queue has been cleared and player stopped"

That's why I commented it doesn't make sense to call it leave while it doesn't actually leave for 247? 
Also I think its worth it to have a stop command to not make the bot leaves directly even if its not 247.

Regarding now playing controller actions I already did it on my fork and added a suggestion on discord if someone is interested in bringing it.
image

**Comment by Saud-97** [comment]
After the clarification do you still think its not worth to PR this command?

**Comment by ItzCuteAkemi** [comment]
/leave does the same thing if /247 is enabled

**Comment by DevAmiyo** [comment]
If this is implemented, `/leave` could be renamed to `/stop` since using `/leave` with 24/7 enabled the bot doesn't "actually leave" so the `/stop` name makes more sense.

**Comment by BioCla** [comment]
I think the matter of naming or renaming is trivial. It's fine to me if you rename the command to `/stop` and change the description to further clarify it's usage

**Comment by Saud-97** [comment]
> /leave does the same thing if /247 is enabled

I honestly think leave should still kick the bot even with 24/7. And the stop command should stop without leaving. Otherwise how would you tell the bot to leave forcefully when 24/7 is enabled? Unless it was an intended behavior to not be able to make the bot leave unless you have disconnect user permission and manually do it.

**Pull request closed without merging**